internal/jsnum/pseudobigint_test.go

```diff
@@ -0,0 +1,39 @@
+package jsnum
+
+import (
+	"strings"
+	"testing"
+
+	"gotest.tools/v3/assert"
+)
+
+func TestParsePseudoBigInt(t *testing.T) {
+	var testNumbers []Number
+	for i := range int64(1e3) {
+		testNumbers = append(testNumbers, Number(i))
+	}
+	for bits := range 53 {
+		testNumbers = append(testNumbers, Number(int64(1<<bits)), Number(int64(1<<bits)-1))
+	}
+
+	t.Run("strip base-10 strings", func(t *testing.T) {
+		for _, testNumber := range testNumbers {
+			for leadingZeros := range 10 {
+				assert.Equal(
+					t,
+					ParsePseudoBigInt(strings.Repeat("0", leadingZeros)+testNumber.String()+"n"),
+					testNumber.String(),
+				)
+			}
+		}
+	})
+
+	// TODO(jakebailey): tests for other bases
```

Comment:
The other test cases here depend on being run in JS where we can convert number to string with multiple bases, something jsnum can't do right now. I might try my hand at that quick, though.